lib/coverband/utils/result.rb

```diff
@@ -100,6 +100,38 @@ def coverage
         Hash[keys.zip(original_result.values_at(*keys))]
       end
 
+      #
+      # Applies the configured groups to the given array of SimpleCov::SourceFile items
+      #
+      def filter_to_groups(files)
+        grouped = {}
+        grouped_files = []
+        grouped_gems = {}
+        gem_lists = []
+        Coverband.configuration.groups.each do |name, filter|
+          if name == 'Gems'
+            gems_files = files.select { |source_file| source_file.filename =~ /#{filter}/ }
```

Comment:
Could this possibly be replaced with `group_by`?

```ruby
grouped_gems = files.select { |source_file| source_file.filename =~ /#{filter}/ }
  .group_by(&:gem_name)
```

Reply:
yeah this is clean